src/createServer.js

```
@@ -1,8 +1,169 @@
+        const end = biteFile.indexOf(boundary, start);
+        const filePart = biteFile.slice(start, end);
+        const headerEnd = filePart.indexOf('\r\n\r\n');
+        let fileBuffer = filePart.slice(headerEnd + 4);
+
+        if (fileBuffer.slice(-2).toString() === '\r\n') {
+          fileBuffer = fileBuffer.slice(0, -2);
+        }
+
+        const fileStream = Readable.from(fileBuffer);
+
+        if (compressType === 'gzip') {
+          const gzip = zlib.createGzip();
+          const newFileName = fileName + '.gzip';
+          const safeFileName = newFileName.replace(/[\r\n]/g, '').trim();
+
+          res.statusCode = 200;
+          res.setHeader('Content-Type', 'application/gzip');
+
+          res.setHeader(
+            'Content-Disposition',
+            `attachment; filename=${safeFileName}`,
+          );
+          fileStream.pipe(gzip);
+          gzip.pipe(res);
+        }
+
+        if (compressType === 'deflate') {
+          const deflate = zlib.createDeflate();
+          const newFileName = fileName + '.deflate';
+
+          res.statusCode = 200;
+          res.setHeader('Content-Type', 'application/deflate');
+
+          res.setHeader(
+            'Content-Disposition',
+            `attachment; filename=${newFileName}`,
+          );
+          fileStream.pipe(deflate);
+          deflate.pipe(res);
+        }
+
+        if (compressType === 'br') {
+          const br = zlib.createBrotliCompress();
+          const newFileName = fileName + '.br';
+
+          res.statusCode = 200;
+          res.setHeader('Content-Type', 'application/br');
+
+          res.setHeader(
+            'Content-Disposition',
+            `attachment; filename=${newFileName}`,
+          );
+          fileStream.pipe(br);
+          br.pipe(res);
+        }
+
+        fileStream.on('error', () => {
+          res.statusCode = 404;
+          res.end('need file');
+        });
+      });
+    }
+  });
```

The code does not return 404 for all non-existent endpoints or methods as required. For example, a POST to `/notfound` or any other undefined route will not return 404. You must add a catch-all handler to return 404 for any undefined route or method.